frontend/junior-1/react.md

```diff
@@ -18,32 +16,31 @@
       </div>
     </OuterComponent>
     ```
+* Что такое условный рендеринг?
+  * В чём проблема следующего кода `<p>{items.length && <El items={items} />}</p>`?
 * Что такое Virtual DOM и зачем он нужен?
-* Что такое `state` и `props` у компонента?
-  * Как обновить `state`? Почему не стоит это делать через прямую запись в `this.state`?
-  * Если передать в `setState` не все свойства, которые до этого уже были в стейте, то будут ли удалены те, что не переданы?
-  * В каких случаях стоит хранить значения в стейте компоненты, а в каких случаях в инстансе?
-  * Как правильно обновлять `state` на основе предыдущего значения `state`? Какие могут при этом возникнуть проблемы?
+* Что такое `props` у компонента?
+* Что такое хуки?
+  * Какие существуют правила при работе с хуками?
+  * Являются ли хуки медленными из-за создания функций на каждом рендере?
```

Comment:
Можешь пояснить этот вопрос, так как я уже второй раз натыкаюсь на него, а понять не могу

Reply:
[дока](https://ru.reactjs.org/docs/hooks-faq.html#are-hooks-slow-because-of-creating-functions-in-render). Это вопрос в контексте сравнения функциональных и классовых компонентов. Я так понимаю он очень часто задавался сообществом, когда в реакте появились хуки. Возможно стоит выпилить?

Reply:
Сам по себе вопрос звучит как то глупо, возможно ввиду своего опыта. Плюс опять же тут идет сравнение с классовыми компонентами, думаю нам не нужен этот вопрос